galaxy_ng/app/api/v3/viewsets/collection.py

```diff
@@ -180,3 +186,64 @@ def get(self, request, *args, **kwargs):
         metrics.collection_artifact_download_failures.labels(status=response.status_code).inc()
         raise APIException('Unexpected response from content app. '
                            f'Code: {response.status_code}.')
+
+
+class CollectionVersionMoveViewSet(ViewSet):
+    permission_classes = GALAXY_PERMISSION_CLASSES + [
+        permissions.IsNamespaceOwner
+    ]
+
+    def move_content(self, request, *args, **kwargs):
+        """Remove content from source repo and add to destination repo.
+
+        Creates new RepositoryVersion of source repo without content included.
+        Creates new RepositoryVersion of destination repo with content included.
+        """
+
+        try:
+            collection_version = CollectionVersion.objects.get(
```

Comment:
At the moment, I'm not sure if a CollectionVersion can exist in multiple repositories, but seems like that would eventually be the case. Does this need to filter on the Repository as well? (To avoid multiple matches)

Reply:
I think eventually a CollectionVersion could exist in more than one Repository, but I don't think this query needs a Repository filter.... this is simply looking up the CollectionVersion whether it exists in a particular Repository or not. Below it is the check to see if it actually exists in the source_repo.

Reply:
Ah, cool.